opensearch-operator/pkg/builders/cluster.go

```diff
@@ -971,3 +973,81 @@ func DataNodesCount(ctx context.Context, k8sClient client.Client, cr *opsterv1.O
 	}
 	return count
 }
+
+func NewServiceMonitor(cr *opsterv1.OpenSearchCluster) *monitoring.ServiceMonitor {
+	ClusterLabels := "opster.io/opensearch-cluster"
+
+	labels := map[string]string{
+		ClusterLabels: cr.Name,
+	}
+	selector := metav1.LabelSelector{
+		MatchLabels: labels,
+	}
+
+	namespaceSelector := monitoring.NamespaceSelector{
+		Any:        false,
+		MatchNames: []string{cr.Namespace},
+	}
+
+	if cr.Spec.General.Monitoring.ScrapeInterval == "" {
+		cr.Spec.General.Monitoring.ScrapeInterval = "30s"
+	}
+	user := monitoring.BasicAuth{}
+
+	monitorUser := cr.Spec.General.Monitoring.MonitoringUserSecret
+	if monitorUser == "" {
```

Comment:
This can be done shorter by only setting a secretName variable in the if and constructing the BasicAuth afterwards:
```
var basicAuthSecret string
if monitorUser == "" {
  basicAuthSecret = cr.Name + "-admin-password"
else {
  basicAuthSecret = cr.Spec.General.Monitoring.MonitoringUserSecret
}
user = monitoring.BasicAuth{...
```